scripts/provisioning-tests

```diff
@@ -122,6 +122,9 @@ dump_rancher_logs()
   echo -e "\n-----RANCHER-LOG-DUMP-END-----"
 }
 
+# Build Rancher
+./scripts/build-server
```

Comment:
Good catch but I think it makes sense to explain more on why we're building it as someone without context may remove it with thoughts like "`./scripts/run` includes build so why are we building it separately?"

Maybe also move it to `run_rancher` function and rename it to something like `build_and_run_rancher` - but that's largely opinion-based so fine to ignore it.

Reply:
addressed as of 29162c8f5